packages/suite/src/middlewares/wallet/walletMiddleware.ts

```diff
@@ -96,6 +101,33 @@ const walletMiddleware =
             api.dispatch(coinmarketBuyActions.dispose());
         }
 
+        if (
+            prevState.router.pathname !== '/accounts/send' &&
+            nextRouter.pathname !== '/accounts/send' &&
+            action.type === WALLET_SETTINGS.SET_BITCOIN_AMOUNT_UNITS
+        ) {
+            const { selectedAccount, settings } = api.getState().wallet;
+            const formattedDraft: FormState = { ...api.dispatch(sendFormActions.getDraft()) };
+
+            if (selectedAccount.account && formattedDraft && formattedDraft.outputs) {
+                const areSatsSelected = settings.bitcoinAmountUnit === PROTO.AmountUnit.SATOSHI;
+                const areSatsSupported = hasNetworkFeatures(selectedAccount.account, 'amount-unit');
+
+                const conversionToUse =
+                    areSatsSelected && areSatsSupported ? amountToSatoshi : formatAmount;
+
+                const selectedNetwork = selectedAccount.network as Network;
```

Comment:
I would use `as` type assertion as last resort, better to verify (e.g. throw an error if selectedNetwork is undefined?) Or if you are sure here, that it's always defined, you could type it with exclamation mark `selectedAccount.network!`

Reply:
not used anymore fa39d3319c7953ea6e6d39c30bc33142d7481bb8